src/main/java/io/nats/client/impl/NatsJetStreamManagement.java

```diff
@@ -324,6 +327,66 @@ private MessageInfo _getMessage(String streamName, MessageGetRequest messageGetR
         }
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public MessageBatchInfo getMessageBatch(String streamName, MessageBatchGetRequest messageBatchGetRequest, Consumer<MessageInfo> consumer) throws IOException, JetStreamApiException {
+        validateNotNull(messageBatchGetRequest, "Message Batch Get Request");
+        validateGtZero(messageBatchGetRequest.getBatch(), "Request batch size");
+
+        CachedStreamInfo csi = getCachedStreamInfo(streamName);
+        if (!csi.allowDirect) {
+            throw new IllegalArgumentException("Allow direct not enabled for stream: " + streamName);
```

Comment:
There are two non-server error conditions:
- either `AllowDirect` is not enabled, in which case we'd get a timeout from the server
- the server is not 2.11+ and doesn't support this feature yet

Currently made both of them `IllegalArgumentException`, but not sure if that's the right fit..
`JetStreamApiException` doesn't fit either as it's for server errors.

Reply:
I think this is fine. Are you going to update NatsJetStreamImpl feature flags to know if the server supports? See lines 48/49 and 71/72. What's a good name for this feature? "direct211"?

Reply:
> Are you going to update NatsJetStreamImpl feature flags to know if the server supports?

Will do :+1: 
Wondering whether those fields will be updated if we were initially connected to a server that supports it, and we reconnect to another server that doesn't?

> What's a good name for this feature? "direct211"?

Yeah, something like that. `directBatchGet211Available`, maybe

Reply:
Added as `directBatchGet211Available`.
And added exceptions for these cases: `JsDirectBatchGet211NotAvailable` and `JsAllowDirectRequired`.